tiledb/api/src/query/read/mod.rs

```diff
@@ -194,21 +194,33 @@ macro_rules! fn_register_callback {
         paste! {
             /// Register a callback to be run on query results
             /// which are written into the provided scratch space.
-            fn $fn<'data, T>(self,
+            fn $fn<T>(self,
                 $(
                     ([< field_ $U:snake >],
                      [< scratch_ $U:snake >]):
-                    (&str, &'data RefCell<QueryBuffersMut<'data, <T as $Callback>::$U>>),
+                    (&str, ScratchStrategy<'data, <T as $Callback>::$U>),
                 )+
                 callback: T
             ) -> TileDBResult<$Builder<'data, T, Self>>
             where
+                Self: Sized,
                 T: $Callback
             {
                 let base = self;
                 $(
-                    let [< arg_ $U:snake >] = RawReadHandle::new(
-                        [< field_ $U:snake >], [< scratch_ $U:snake >]);
+                    let [< arg_ $U:snake >] = {
+                        let field = [< field_ $U:snake >];
+                        match [< scratch_ $U:snake >] {
+                            ScratchStrategy::AttributeDefault => unimplemented!(),
```

Comment:
This shouldn't be that hard to implement - get the array ref, get the schema ref, look up the attributes by name, call the "query allocator" function.